pkg/server/api_registration.go

```diff
@@ -159,6 +159,10 @@ func (i *InventoryServer) writeMachineInventoryCloudConfig(conn *websocket.Conn,
 		registration.Spec.Config = &elementalv1.Config{}
 	}
 
+	if registration.Status.RegistrationURL == "" {
```

Comment:
Here we are getting the full config after a registration process: the issue instead is about the registration data, which are retrieved via a plan HTTP GET request and returned by unauthenticatedResponse().
So, this would not help the SeedImage build scenario.

Reply:
Good point @fgiudici , thanks much for the review, I just found where this data was retrieved and I did not really consider other potential paths. Gonna try come with a small refactor so there is a single sanity check needed and less duplicated code.